isis/tests/BasisFunctionTests.cpp

```diff
@@ -0,0 +1,142 @@
+#include <gtest/gtest.h>
+#include <QString>
+#include "BasisFunction.h"
+#include "IException.h"
+
+TEST(BasisFunction, Initialization)
+{
+  std::string name("basis1");
+  Isis::BasisFunction testBasis("basis1", 3, 3);
+
+  EXPECT_EQ(testBasis.Coefficients(), 3);
+  EXPECT_EQ(testBasis.Variables(), 3);
+  EXPECT_STREQ(testBasis.Name().toStdString().c_str(), name.c_str());
+}
+
+TEST(BasisFunction, Evaluation)
+{
+  Isis::BasisFunction testBasis("basis", 3, 3);
+  double expectedOutput = 2.5*3.0 + 3.2 + 2.0;
```

Comment:
Why not just have the result here (i.e. 12.7)?

Reply:
I just did that out of laziness. I just changed them to the simplified versions so everything should be good now.